csharp/src/Ice/Reference.cs

```diff
@@ -1749,9 +1492,13 @@ private void ClearConnection(Connection connection)
                 {
                     properties[$"{prefix}.LocatorCacheTimeout"] = locatorCacheTimeout.ToPropertyValue();
                 }
-                if (_preferNonSecure is bool preferNonSecure)
+                if (!PreferExistingConnection)
```

Comment:
Only create the property if different from communicator's default?

Reply:
We do the same for `CacheConnection` https://github.com/zeroc-ice/ice/blob/f1ed3c9e2b1a26c0a9ab766921b48a086d338e38/csharp/src/Ice/Reference.cs#L584

Reply:
Ok right, like for CacheConnection, there's no communicator `Ice.Default.PreferExistingConnection` property.

Reply:
I think it makes sense for the default `CacheConnection` to be always true. 

It's less obvious that the default `PreferExistingConnection` should be always true. It makes it harder to write a client that reverts back to the first / primary endpoint as soon as possible.

Reply:
I will add `Ice.Default.PreferExistingConnection` with default to false

Reply:
The default should probably be true for compatibility with Ice 3.x.